src/Common/tests/System/Net/Http/LoopbackServer.cs

```diff
@@ -454,32 +463,80 @@ public override async Task<HttpRequestData> HandleRequestAsync(HttpStatusCode st
+
+                // Convert header lines to key/value pairs
+                // Skip first line since it's the status line
+                foreach (var line in headerLines.Skip(1))
+                {
+                    int offset = line.IndexOf(':');
+                    string name = line.Substring(0, offset);
+                    string value = line.Substring(offset + 1).TrimStart();
+                    requestData.Headers.Add(new HttpHeaderData(name, value));
+                }
+
+                if (requestData.Method != "GET")
+                {
+                    if (requestData.GetHeaderValueCount("Content-Length") != 0)
+                    {
+                        int contentLength = Int32.Parse(requestData.GetSingleHeaderValue("Content-Length"));
+                        byte[] buffer = new byte[contentLength];
+                        int offset = 0;
+
+                        while (contentLength != 0)
+                        {
+                            int length = await connection.Reader.BaseStream.ReadAsync(buffer, offset, contentLength);
+                            contentLength -= length;
+                            offset += length;
+                        }
+
+                        requestData.AddBodyData(buffer);
+                    }
+                    else if (requestData.GetHeaderValueCount("Transfer-Encoding") != 0 && requestData.GetSingleHeaderValue("Transfer-Encoding") == "chunked")
+                    {
+                        while (true)
+                        {
+                            string chunkHeader = await connection.Reader.ReadLineAsync();
+                            if (chunkHeader == null)
+                            {
+                                // Some tests do not send complete body.
+                                break;
+                            }
+                            int contentLength = int.Parse(chunkHeader, System.Globalization.NumberStyles.HexNumber);
+                            if (contentLength == 0)
+                            {
+                                // Last chunk
+                                break;
+                            }
+                            char[] buffer = new char[contentLength];
+                            int offset = 0;
+                            while (contentLength != 0)
+                            {
+                                int length = await connection.Reader.ReadAsync(buffer, offset, contentLength);
+                                contentLength -= length;
+                                offset += length;
+                            }
+
+                            requestData.AddBodyData(Encoding.GetEncoding("ASCII").GetBytes(buffer).AsSpan());
+                            await connection.Reader.ReadLineAsync();
+                        }
+                    }
+                    else
+                    {
+                        await connection.Drain();
```

Comment:
Is this needed? Seems like this should be an error.

Reply:
This was attempt to mimic actual handler. I removed it for now. We can revisit this later as we convert more tests.